services/server/src/server/services/utils/signature-util.ts

```diff
@@ -9,16 +9,21 @@ export interface SignatureData {
 export function extractSignaturesFromAbi(abi: JsonFragment[]): SignatureData[] {
   const signatures: SignatureData[] = [];
 
-  const iface = new Interface(abi);
-
-  iface.fragments.forEach((fragment) => {
-    switch (fragment.type) {
-      case "function":
-      case "event":
-      case "error":
-        signatures.push(getSignatureData(fragment));
+  try {
+    for (const item of abi) {
+      const fragment = Fragment.from(item);
+      switch (fragment.type) {
+        case "function":
+        case "event":
+        case "error":
+          signatures.push(getSignatureData(fragment));
+      }
     }
-  });
+  } catch (error) {
+    throw new Error(
+      "Failed to extract signatures from ABI due to an invalid fragment",
+    );
+  }
```

Comment:
This should not throw, or if it does I don't see it being caught anywhere. If there's an invalid type e.g. when it's a library contract ABI we should simply skip. Right now this throws here https://github.com/argotorg/sourcify/blob/6bfbf5f848e869451a294475797efd1017f1fb8b/services/server/src/server/services/storageServices/SourcifyDatabaseService.ts#L913-L914